### PR TITLE
Stop widget

### PR DESCRIPTION
**Pull request by AgeX21**
In this PR, the following has been done : 

- There is a new stop widget in the dashboard to display the upcoming stops and the total number of stops remaining
- Adjusted the display of all the widgets to make the dashboard prettier

### CODE REVIEW

File: app/src/main/java/com/github/se/wanderpals/ui/screens/dashboard/DashboardStopWidget.kt [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+            }
+      }
+}
+
+/**
+ * Composable function to display a single stop item in the dashboard widget
+ * @param stop Stop object to display
+ */
+@Composable
+fun StopItem(stop: Stop) {
```
Comment (ForestDope) [inline app/src/main/java/com/github/se/wanderpals/ui/screens/dashboard/DashboardStopWidget.kt:172]: Rename the composable as there is another composable called StopItem used in the stops list and the agenda.

File: app/src/main/java/com/github/se/wanderpals/ui/screens/dashboard/DashboardStopWidget.kt [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+                                      .padding(horizontal = 8.dp, vertical = 4.dp),
+                              style =
+                                  TextStyle(
+                                      color = MaterialTheme.colorScheme.primary,
+                                      fontWeight = FontWeight.Bold))
+                        }
+
+                    Spacer(modifier = Modifier.padding(4.dp))
+
+                    // Latest expenses
```
Comment (ForestDope) [inline app/src/main/java/com/github/se/wanderpals/ui/screens/dashboard/DashboardStopWidget.kt:130]: Changes this comment

File: app/src/main/java/com/github/se/wanderpals/ui/screens/dashboard/DashboardStopWidget.kt [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+@Composable
+fun DashboardStopWidget(viewModel: DashboardViewModel, onClick: () -> Unit = {}) {
+  val stops by viewModel.stops.collectAsState()
+  val sortedStops =
+      stops
+          .sortedBy { LocalDateTime.of(it.date, it.startTime) }
+          .filter {
+            LocalDateTime.of(it.date, it.startTime)
+                .plusMinutes(it.duration.toLong())
+                .isAfter(LocalDateTime.now())
```
Comment (ForestDope) [inline app/src/main/java/com/github/se/wanderpals/ui/screens/dashboard/DashboardStopWidget.kt:59]: You should display the coming stops instead not the ones that are over.

**Review by ForestDope — CHANGES_REQUESTED** [review]
Overall good, especially the UI but please update the comments, the composable name and the logic for which stop to display.

File: app/src/main/java/com/github/se/wanderpals/ui/screens/dashboard/DashboardStopWidget.kt (hunk 1 above)
Comment (AgeX21) [reply app/src/main/java/com/github/se/wanderpals/ui/screens/dashboard/DashboardStopWidget.kt:172]: Done

File: app/src/main/java/com/github/se/wanderpals/ui/screens/dashboard/DashboardStopWidget.kt (hunk 2 above)
Comment (AgeX21) [reply app/src/main/java/com/github/se/wanderpals/ui/screens/dashboard/DashboardStopWidget.kt:130]: Done

**Review by ForestDope — APPROVED** [review]
Requested changes are present, nice !

**Review by ForestDope — APPROVED** [review]
Good job !

**Pull request merged**